include/picongpu/particles/atomicPhysics/electronDistribution/LocalHistogramField.hpp

```diff
@@ -0,0 +1,130 @@
+     * interleave
+     */
+    template<bool printOnlyOverSubscribed>
+    struct PrintHistogramToConsole
+    {
+        //! cpu version
+        template<typename T_Acc, typename T_Histogram>
+        HDINLINE auto operator()(
+            T_Acc const&,
+            T_Histogram const& histogram,
+            pmacc::DataSpace<picongpu::simDim> superCellIdx) const
+            -> std::enable_if_t<std::is_same_v<alpaka::Dev<T_Acc>, alpaka::DevCpu>>
+        {
+            constexpr uint32_t numBins = T_Histogram::numberBins;
+
+            std::cout << "histogram " << superCellIdx.toString(",", "[]");
+            std::cout << " base=" << histogram.getBase();
+            std::cout << " numBins=" << T_Histogram::numberBins;
+            std::cout << " maxE=" << T_Histogram::maxEnergy;
+            std::cout << " overFlow: w0=" << histogram.getOverflowWeight() << std::endl;
+
+            float_X centralEnergy;
+            float_X binWidth;
+
+            for(uint32_t i = 0u; i < numBins; i++)
+            {
+                if constexpr(printOnlyOverSubscribed)
+                {
+                    if(histogram.getBinWeight0(i) >= histogram.getBinDeltaWeight(i))
+                        continue;
+                }
+
+                // binIndex
+                std::cout << "\t " << i;
+
+                // central bin energy [eV] and binWidth [eV]
+                centralEnergy = histogram.getBinEnergy(i);
+                binWidth = histogram.getBinWidth(i);
+
+                std::cout << "(" << centralEnergy - binWidth / 2._X << ", " << centralEnergy + binWidth / 2._X
+                          << "] :";
+
+                // bin data, [w0, DeltaW, DeltaEnergy, binOverSubscribed]
+                std::cout << " [w0, Dw, DE]: [";
+                std::cout << histogram.getBinWeight0(i) << ", ";
+                std::cout << histogram.getBinDeltaWeight(i) << ", ";
+                std::cout << histogram.getBinDeltaEnergy(i) << "]";
+                std::cout << std::endl;
+            }
+        }
+
+        //! gpu verison, does nothing
+        template<typename T_Acc, typename T_Histogram>
+        HDINLINE auto operator()(
+            T_Acc const&,
+            T_Histogram const& histogram,
+            pmacc::DataSpace<picongpu::simDim> superCellIdx) const
+            -> std::enable_if_t<!std::is_same_v<alpaka::Dev<T_Acc>, alpaka::DevCpu>>
+        {
+        }
+    };
+
+    /** holds a gridBuffer of the per-superCell localHistograms for atomicPhysics
+     *
+     * @attention histograms are uninitialized upon creation of the field, use .getDeviceBuffer().setValue() to init
+     */
+    template<typename T_Histogram, typename T_MappingDescription>
+    struct LocalHistogramField : SuperCellField<T_Histogram, T_MappingDescription, false /*no guards*/>
+    {
+        using Histogram = T_Histogram;
+
+    private:
+        /// @todo should these be private?
```

Comment:
QUESTION: What's the status of this?

Reply:
I would keep them private, but psychocoder's opinion differed in the past. therefore marked for discussion

Reply:
removed comment, will keep the string private to prevent name changes after creation